### PR TITLE
forked suggestions by kaduk with further edits

### PR DESCRIPTION
**Pull request by royfielding**
Replaces #870 by @kaduk (minus the bits already committed) with adjustments by me.

### CODE REVIEW

File: draft-ietf-httpbis-semantics-latest.xml [hunk 1]
```diff
@@ -1116,6 +1118,7 @@ Content-Type: text/plain
 <t>
    The origin server for an "https" URI is identified by the
    <x:ref>authority</x:ref> component, which includes a host identifier
+   (<xref target="RFC3986" x:fmt="," x:sec="3.2.2"/>)
    and optional port number (<xref target="RFC3986" x:fmt="," x:sec="3.2.2"/>).
```
Comment (reschke) [inline draft-ietf-httpbis-semantics-latest.xml:1122]:
```suggestion
   and optional port number (<xref target="RFC3986" x:fmt="," x:sec="3.2.3"/>).
```

File: draft-ietf-httpbis-semantics-latest.xml [hunk 2]
```diff
@@ -1745,7 +1748,8 @@ Example-Dates: "Sat, 04 May 1996", "Wed, 14 Sep 2005"
    that allows both unquoted (token) and quoted (quoted-string) syntax for
    a parameter value (<xref target="parameter"/>). Use of common syntax
    allows recipients to reuse existing parser components. When allowing both
-   forms, the meaning of a parameter value ought to be the same whether it
+   forms, the meaning of a parameter value (with no internal commas)
```
Comment (reschke) [inline draft-ietf-httpbis-semantics-latest.xml:1751]: What is the "(with no internal commas)" about?

**Review by reschke — CHANGES_REQUESTED** [review]

File: draft-ietf-httpbis-semantics-latest.xml (hunk 2 above)
Comment (royfielding) [reply draft-ietf-httpbis-semantics-latest.xml:1751]: no idea, will remove this part

File: draft-ietf-httpbis-semantics-latest.xml [hunk 3]
```diff
@@ -1745,7 +1748,8 @@ Example-Dates: "Sat, 04 May 1996", "Wed, 14 Sep 2005"
...
+   ought to be the same whether it
```
Comment (royfielding) [inline draft-ietf-httpbis-semantics-latest.xml:1752]:
```suggestion
   forms, the meaning of a parameter value ought to be the same whether it
```

File: draft-ietf-httpbis-semantics-latest.xml (hunk 2 above)
Comment (kaduk) [reply draft-ietf-httpbis-semantics-latest.xml]: You can't use the token syntax if there are commas in the parameter value.  If there are such "internal commas" in the parameter value, then only the quoted-string syntax can be used for that particular value, regardless of whether or not the parameter in question allows both token and quoted-string form.
So this statement about "meaning ought to be the same whether it was received as [...]" is nonsensical for a parameter value that has internal commas.

**Review by reschke — APPROVED** [review]

File: draft-ietf-httpbis-semantics-latest.xml (hunk 2 above)
Comment (reschke) [reply draft-ietf-httpbis-semantics-latest.xml]: No, the statement is true for syntactically valid tokens and quoted-strings.

**Review by mnot — APPROVED** [review]

**Pull request merged**